### PR TITLE
Implement `seal_rent_params`

### PR DESCRIPTION
**Pull request by cmichi**
Needed for https://github.com/paritytech/ink/issues/539.

We haven't yet finalized this API as stable, hence I've put it behind a feature flag for now.

Examples will follow in a separate PR.

### CODE REVIEW

**Comment by Robbepop** [comment]
I have had a discussion with @athei where he told me that changes to the API are already in the pipeline about a less expressive but still "good enough" API using a single `seal_rent_deposit` API to query the rent deposit with the current storage used for the contract. Note that this API would be called multiple times during a call. However, the good part is that this would not require manually calculating storage fees on the contract side.

**Comment by athei** [comment]
Yes. I talked to @cmichi about that. I will look into the next iteration of the `seal_rent_deposit` API soon.

File: crates/env/src/engine/on_chain/ext.rs [hunk 1]
```diff
@@ -589,6 +593,19 @@ pub fn set_rent_allowance(value: &[u8]) {
     unsafe { sys::seal_set_rent_allowance(Ptr32::from_slice(value), value.len() as u32) }
 }
 
+pub fn rent_params(output: &mut &mut [u8]) {
```
Comment (athei) [inline crates/env/src/engine/on_chain/ext.rs:596]: Need to be put in a
```rust
#[link(wasm_import_module = "__unstable__")]
extern "C" { }
```
block.

Comment (Robbepop) [reply crates/env/src/engine/on_chain/ext.rs:596]: why not just `"unstable"` btw?

Comment (athei) [reply crates/env/src/engine/on_chain/ext.rs:596]: I use it to indicate that it has some special semantic and is not only a module name.

**Comment by cmichi** [comment]
@athei The comment is implemented, do you have anything else?

**Comment by athei** [comment]
It was discussed [here](https://github.com/paritytech/ink/pull/792#issuecomment-845818188) that no `ink-unstable` feature should be added. Maybe it should be removed.

**Comment by cmichi** [comment]
> It was discussed [here](https://github.com/paritytech/ink/pull/792#issuecomment-845818188) that no `ink-unstable` feature should be added. Maybe it should be removed.

I get a bit sweaty when thinking about making the current rent API accessible without unstable flag. The reason is that in my mind it's pretty certain that we will have another iteration over the current status and not having it behind `ink-unstable` would mean that we e.g. have to mention it in the next rc changelog. Wdyt @Robbepop?

**Comment by cmichi** [comment]
I thought a bit more about it and came to the conclusion that it's best to remove the unstable feature for rent stuff as well. My reasoning is that we are unstable in the rc state anyway and I also want to add examples soon and making them only available on unstable would make them more complicated. Also we might get community input by not hiding this behind a feature flag.

**Review by athei — APPROVED** [review]

**Pull request merged**